discovery/azure/azure.go

```diff
@@ -145,18 +145,22 @@ func (c *SDConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	if err != nil {
 		return err
 	}
-	if err = validateAuthParam(c.SubscriptionID, "subscription_id"); err != nil {
+	err = validateAuthParam(c.SubscriptionID, "subscription_id")
+	if err != nil {
```

Comment:
Why? Combined `if err` should be preferred for sure.

Maybe it complained about reusing variable? All of those should definitely be new shadowed variables to ensure local scope and no side effects e.g.

```
if err = some(); err != nil {
```

Should be 

```
if err := some(); err != nil {
```

... and probably that's why go-critic is complaing?

So all of those should be: 

```suggestion
	if err := validateAuthParam(c.SubscriptionID, "subscription_id"); err != nil {
```